phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/engines/EnginesFilter.java

```diff
@@ -64,27 +106,42 @@ public EnginesFilter(EngineCategoryDTO engineCategory, EngineSubCategoryDTO engi
      * @param engineVersionDTO The engine version to be checked
      * @return True if the engine version is installed, false otherwise
      */
-    private boolean isInstalled(EngineVersionDTO engineVersionDTO) {
+    private boolean isInstalled(EngineCategoryDTO engineCategory, EngineSubCategoryDTO engineSubCategory,
+            EngineVersionDTO engineVersionDTO) {
         return Paths.get(enginesPath, engineCategory.getName().toLowerCase(), engineSubCategory.getName(),
                 engineVersionDTO.getVersion()).toFile().exists();
     }
 
-    @Override
-    public boolean test(EngineVersionDTO engineVersion) {
-        return engineVersion.getVersion().toLowerCase().contains(searchTerm.toLowerCase())
-                && ((this.showInstalled && isInstalled(engineVersion))
-                        || (this.showNotInstalled && !isInstalled(engineVersion)));
-    }
-
-    public void setSearchTerm(String searchTerm) {
-        this.searchTerm = searchTerm;
-    }
-
-    public void setShowInstalled(boolean showInstalled) {
-        this.showInstalled = showInstalled;
+    /**
+     * Creates a new filter predicate for a {@link EngineCategoryDTO} and {@link EngineSubCategoryDTO}.
+     * This predicate then accepts a {@link EngineVersionDTO} object and returns true if the given object fulfills the filter predicate and false otherwise
+     *
+     * @param engineCategory    The engine category
+     * @param engineSubCategory The engine sub category
+     * @return A new filter predicate
+     */
+    public Predicate<EngineVersionDTO> createFilter(EngineCategoryDTO engineCategory,
+            EngineSubCategoryDTO engineSubCategory) {
+        return engineVersion -> {
+            return searchTerm
+                    .map(searchTerm -> engineVersion.getVersion().toLowerCase().contains(searchTerm.toLowerCase()))
+                    .orElse(true)
```

Comment:
Can this be formatted differently?

Reply:
How would you like it to look?
This is how maven formats it

Reply:
I think it should respect manual line breaks.

I'm not quite sure but currently it's a bit hard to see what belongs together. Maybe somehow like this:
```Java
return searchTerm.map(
    searchTerm -> engineVersion.getVersion().toLowerCase().contains(searchTerm.toLowerCase())).orElse(true)
    && ((this.showInstalled.getValue() && isInstalled(engineCategory, engineSubCategory, engineVersion))
    || (this.showNotInstalled.getValue() && !isInstalled(engineCategory, engineSubCategory, engineVersion))
);

Reply:
Maybe the better solution would be to create boolean variables for the different parts of the expression.